### DISCUSSION TITLE
Global Settings Control

### DISCUSSION THREAD

**Discussion by duczz** [rustdesk/rustdesk#3093] [General]
It would be great when we can control the rustdesk settings (z.b. directaccess) via regkey, copy config files or install parameter.
Currently rustdesk overwrite the copyed new settingsfile on startup

**Comment by Geexirooz** [comment]
I have the same issue.
I have to roll out Rustdesk to all of my users computers and I don't want them to change any configuration at all.

I installed Rustdesk through Puppet and put desired configuration files in place and it perfectly works on MacOS but when it comes to Ubuntu machines (22.04) it does not work as intended (maybe there is the same issue with Windows as well -> I haven't tried yet).
I put configuration files under `/root/.config/rustdesk` on Ubuntu and when you run rustdesk as root it seems fine but when an ordinary user runs Rustdesk, it generates new configuration files under the user's profile (`/home/user/.config/rustdesk`) which ignores my desired configuration and tries to connect to Rustdesk public servers.

How can we fix this issue and force Rustdesk to read root's config files ("Global" configuration files)?

BTW, I'm using Nightly Build.


@rustdesk can you help us in here please?

**Reply by rustdesk** [reply]
> on Ubuntu and when you run rustdesk as root it seems fine but when an ordinary user runs Rustdesk, it generates new configuration files under the user's profile (`/home/user/.config/rustdesk`) which ignores my desired configuration and tries to connect to Rustdesk public servers.

@Geexirooz  I still do not understand how do you run as user? I just tried modify customer server under `/root/.config`, and `sudo service rustdesk restart`, it works as expected.

**Reply by Geexirooz** [reply]
@rustdesk  Sorry for late response. You're right I was doing something wrong.
I have the same question for Windows deployment. I have used the silent install feature and deployed it all over my windows 10 machines which are joined to my domain. The question is where should I put my desired configuration files so that when a domain user logs into these machines their Rustdesk is using the desired configuration.
I put it under Domain administrator's profile and local administrator's profile but none of them worked (I thought it might be working as it worked with root user in Linux). 
I have seen the document which is suggesting renaming the exe file but I think that's true for a portable Rustdesk not the one I'm talking about. Is there any way that I can install Rustdesk in my computer nodes and put in a global configurations that no user can change it and they can only use it?
Just in case you want to see how I'm deploying, here is my Puppet configuration for Rustdesk:
```
# Install Rustdesk
class software::rustdesk {
  case $::operatingsystem {
    'Windows':  { include software::rustdesk::windows }
    default: { include software::rustdesk::ubuntu }
  }
}

# Linux install
class software::rustdesk::ubuntu {
  package {
    'rustdesk':
      ensure  => installed,
      require => Apt::Source['local-repo'];
  }

  service {
    'rustdesk':
      ensure  => 'running',
      enable  => true,
      require => Package['rustdesk'];
  }

  file {
    '/etc/lightdm/lightdm.conf.d/rustdesk-root-permission.conf':
      content => "[SeatDefaults]\ngreeter-setup-script=xhost +SI:localuser:root",
      require => Package['rustdesk'];

    '/root/.config/rustdesk/RustDesk2.toml':
      content => template('software/RustDesk2.toml.erb'),
      notify  => Service['rustdesk'],
      require => Package['rustdesk'];
  }
}

# Windows install
class software::rustdesk::windows {
  package {
    'Rustdesk':
      source          => "${::winrepo}\\rustdesk\\rustdesk-1.2.2-x86_64.exe",
      install_options => ['--silent-install'],
      provider        => 'windows';
  }

#Where should I put this config file????????
  file {
    'c:/Users/Administrator/AppData/Roaming/RustDesk/config/RustDesk2.toml':
      content => template('software/RustDesk2.toml.erb'),
      require => Package['Rustdesk'];
  }

}
```
As you can see I'm using `RustDesk2.toml.erb` as a template config.

**Reply by Geexirooz** [reply]
@rustdesk Any idea about this?

**Reply by Geexirooz** [reply]
@rustdesk 
I really need help on this issue, thanks for your time in advance

**Comment by Geexirooz** [comment]
@rustdesk Thank you for taking an action but I would call this an issue not a discussion because Rustdesk is not working as intended which is definition of an issue I believe.

**Comment by rustdesk** [comment]
rustdesk is not run as service?

**Reply by duczz** [reply]
It is running as service

**Comment by rustdesk** [comment]
Change back to issue https://github.com/rustdesk/rustdesk/discussions/3101

**Comment by rustdesk** [comment]
close this since we have custom client feature.

https://twitter.com/rustdesk/status/1788905463678951787?t=G0DEhYLYYfALeiUoUFwY7A&s=19
